packages/nimble-components/src/wafer-map/tests/hover-handler.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { zoomIdentity } from 'd3-zoom';
+import { tableFromArrays } from 'apache-arrow';
+import { html } from '@microsoft/fast-element';
+import { HoverHandler } from '../modules/experimental/hover-handler';
+import { WaferMapOriginLocation } from '../types';
+import {
+    getDataManagerMockForHover,
+    getScaleQuantile,
+    getWaferMapMockHover
+} from './utilities';
+import type { WaferMap } from '..';
+import type { DataManager } from '../modules/data-manager';
+import { processUpdates } from '../../testing/async-helpers';
+import { Fixture, fixture } from '../../utilities/tests/fixture';
+
+async function setup(): Promise<Fixture<HTMLDivElement>> {
+    return fixture<HTMLDivElement>(html`<div></div>`);
+}
+
+describe('HoverHandler', () => {
+    let element: HTMLDivElement;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let hoverHandler: HoverHandler;
+    let waferMock: WaferMap;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+        waferMock = getWaferMapMockHover(
+            tableFromArrays({
+                colIndex: Int32Array.from([1]),
+                rowIndex: Int32Array.from([1]),
+                value: Float64Array.from([1])
+            }),
+            zoomIdentity,
+            WaferMapOriginLocation.bottomLeft,
+            undefined,
+            getDataManagerMockForHover(
+                { left: 0, right: 0, top: 0, bottom: 0 },
+                getScaleQuantile([1, 10], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+                getScaleQuantile([1, 10], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+            ) as DataManager
+        ) as WaferMap;
+        hoverHandler = new HoverHandler(waferMock);
+        element.addEventListener('mousemove', event => hoverHandler.mousemove(event));
```

Comment:
Assuming you made the changes that I suggested in the `HoverHandler` implementation, I would think this would have to change such that the method that is creating the `waferMock` would also handle creating the `hoverHandler` instance, and be responsible for calling its `connect` implementation.

Reply:
Thinking through this further, and after discussing with @jattasNI, we agree that the use of mocks to test this functionality seems unnecessarily convoluted. Is there difficulty in hosting an actual `WaferMap` instance in the DOM, with some test dies, and then issuing a `mousemove` event on the `WaferMap` itself?

Reply:
the experimental hover handler needs a data manager instance which is not implemented yet. if we create a wafer instance, I will still need to replace the existing data manager with a mock and remove the existing event handlers. These mocks are temporary to not increase the scope of this PR with the implementation for the data manager

Reply:
At the very least we need to create an issue (or link to an existing one) and then link to that issue in the test code and explain that we intend to change the implementation. As is, it feels like this is an incomplete way to mock out dependencies. Having what seems like an arbitrary div in the DOM to stand-in for the wafer-map to issue events through seems janky. We should make it clear that the end goal is to just use a real `WaferMap` instance.